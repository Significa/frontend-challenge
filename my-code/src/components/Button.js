import React from 'react';

//css
import {
	ButtonPrimaryStyle,
	ButtonPrimaryActiveStyle,
	ButtonLinkStyle,
	ButtonCardStyle
} from '../Styles/styles';


////
export const ButtonPrimary = (props) => {
	if (props.isActive) {
		return (
			<ButtonPrimaryActiveStyle type='button' onClick={props.onClick} data-testid='buttonPrimaryTrue'>
				{props.icon && <i className={props.icon} aria-hidden='true' />}
				{props.children}
			</ButtonPrimaryActiveStyle>
		);
	} else {
		return (
			<ButtonPrimaryStyle type='button' onClick={props.onClick} data-testid='buttonPrimaryFalse'>
				{props.icon && <i className={props.icon} aria-hidden='true' />}
				{props.children}
			</ButtonPrimaryStyle>
		);
	}
};

export const ButtonLink = (props) => (
	<ButtonLinkStyle type='button' onClick={props.onClick} data-testid='buttonLink'>
		{props.icon && <i className={props.icon} aria-hidden='true' />}
		{props.children}
	</ButtonLinkStyle>
);

export const ButtonCard = (props) => (
	<ButtonCardStyle type='button' onClick={props.onClick} data-testid='buttonCard'>
		{props.icon && <i className={props.icon} aria-hidden='true' data-testid='buttonCardIcon' />}
		{props.children}
	</ButtonCardStyle>
);
