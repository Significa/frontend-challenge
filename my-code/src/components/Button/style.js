import sc, { css } from 'styled-components'
import { theme } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.button`
  display: flex;
  align-items: center;
  /* Account for border */
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: ${theme('greyLight')};
  border: 1px solid ${theme('greyMid')};
  border-radius: 4px;
  cursor: pointer;
  transition-timing-function: ${theme('transitionTiming')};
  transition-duration: ${theme('transitionDuration')};
  transition-property: color, border-color, background-color;
  font-family: inherit;
  ${text4}
  font-weight: 500;

  &:focus,
  &:hover {
    border-color: ${theme('mainRed')};
    color: ${theme('greyWhite')};
  }

  ${({ active }) =>
    active &&
    css`
      border-color: ${theme('mainRed')};
      background-color: ${theme('mainRed')};
      color: ${theme('greyWhite')};
    `}
`

export const IconWrapper = sc.div`
  width: 16px;
  height: 16px;
  margin-right: 12px;
  display: flex;
  align-items: center;
`
