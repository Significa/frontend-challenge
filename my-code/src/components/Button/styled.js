// @ flow
import styled from "styled-components"

const RegularButton = styled.button`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;

  background: none;
  border: 1px solid ${p => p.theme.color.grey.lightgrey};
  border-radius: 4px;
  color: ${p => p.theme.color.grey.lightgrey};
  font-family: ${p => p.theme.font.family};
  font-size: 1em;
  font-weight: bold;
  line-height: 19px;
  letter-spacing: 0.17;
  padding: 12px;
  outline: none;

  transition: ${p => p.theme.transition};

  &:hover {
    border: 1px solid ${p => p.theme.color.main.red};
    color: ${p => p.theme.color.grey.white};
    outline: none;
  }

  &:active {
    background-color: ${p => p.theme.color.main.red};
    outline: none;
  }
`
export default RegularButton
