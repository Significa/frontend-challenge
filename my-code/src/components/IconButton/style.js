import sc from 'styled-components'
import { theme } from 'styled-tools'

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = sc.button`
  height: 40px;
  outline: none;
  border: none;
  padding: 0;
  background: none;
  color: ${theme('greyLight')};
  cursor: pointer;
  transition:
    color
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  display: flex;
  align-items: center;
  justify-content: center;

  &:focus,
  &:hover {
    color: ${theme('greyWhite')};
  }
`
