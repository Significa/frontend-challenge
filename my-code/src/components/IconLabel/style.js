import sc from 'styled-components'
import { theme, prop } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.div`
  display: flex;
  align-items: stretch;
  height: 40px;
  margin-right: 16px;
  margin-bottom: 8px;
`

export const IconWrapper = sc.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${prop('color')};
  border-radius: 4px 0 0 4px;
  padding: 8px;
  min-width: 40px;
  box-sizing: border-box;
`

export const TextWrapper = sc.div`
  border: 1px solid ${theme('greyMid')};
  border-left: none;
  /* Account for border */
  padding: 8px 7px 8px 8px;
  border-radius: 0 4px 4px 0;
  ${text4}
  font-weight: 500;
`
