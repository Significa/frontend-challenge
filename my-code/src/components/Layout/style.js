import sc from 'styled-components'

// eslint-disable-next-line import/prefer-default-export
export const Outer = sc.div`
  display: flex;
  flex-direction: column;
  max-width: 1228px;
  margin: 0 auto;
  padding: 24px;
  min-height: 100vh;
  box-sizing: border-box;
`
