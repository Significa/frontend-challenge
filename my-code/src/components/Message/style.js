import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text2, text4 } from '../../styles/text'

import illustration from '../../assets/illustration-empty-state@2x.png'

export const Center = sc.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`

export const Illustration = sc.img.attrs({
  src: illustration,
  alt: ''
})`
  object-fit: contain;
  width: 400px;
  max-width: 100%;
  height: 200px;
`

export const Title = sc.span`
  ${text2}
  margin-top: 16px;
`

export const Subtitle = sc.span`
  ${text4}
  color: ${theme('greyLight')};
  margin-top: 8px;
`
