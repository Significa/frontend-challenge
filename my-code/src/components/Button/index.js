// @flow
import React from "react"
import RegularButton from "./styled"
// @flow
import Heart from "../Icons/Heart"
import Wrapper from "../../layout/Wrapper"
import { Text100 } from "../Typography/styled"

type PropsType = {
  text: string
}

const Button = (props: PropsType) => {
  const { text } = props
  return (
    <RegularButton>
      <Wrapper>
        <Heart />
        <Text100 grey>{text}</Text100>
      </Wrapper>
    </RegularButton>
  )
}

export default Button
