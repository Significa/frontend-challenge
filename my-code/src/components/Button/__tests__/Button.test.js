import React from 'react'
import { shallow } from 'enzyme'

import Button from '../'
import { Wrapper, IconWrapper } from '../style'

function MockIcon() {
  return null
}

describe('Button', () => {
  it('renders correctly', () => {
    const button = shallow(<Button onClick={() => {}}>button label</Button>)
    expect(button).toMatchSnapshot()
  })

  it('renders correctly with icon', () => {
    const button = shallow(
      <Button onClick={() => {}} icon={MockIcon}>
        icon button label
      </Button>
    )
    expect(button).toMatchSnapshot()
  })

  it('renders active correctly', () => {
    const button = shallow(
      <Button onClick={() => {}} active>
        active button text
      </Button>
    )
    expect(button).toMatchSnapshot()
  })

  it('calls onClick when clicked', () => {
    const mockCallback = jest.fn()
    const button = shallow(<Button onClick={mockCallback}>button label</Button>)
    button.simulate('click')
    expect(mockCallback.mock.calls.length).toBe(1)
  })

  describe('Wrapper', () => {
    it('renders correctly', () => {
      const tree = shallow(<Wrapper />)
      expect(tree).toMatchSnapshot()
    })

    it('renders correclty when active', () => {
      const tree = shallow(<Wrapper active />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('IconWrapper', () => {
    it('renders correctly', () => {
      const tree = shallow(<IconWrapper />)
      expect(tree).toMatchSnapshot()
    })
  })
})
