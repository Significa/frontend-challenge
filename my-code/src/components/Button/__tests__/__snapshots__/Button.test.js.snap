// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button IconWrapper renders correctly 1`] = `
.c0 {
  width: 16px;
  height: 16px;
  margin-right: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;

exports[`Button Wrapper renders correclty when active 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: theme.greyLight;
  border: 1px solid theme.greyMid;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition-timing-function: theme.transitionTiming;
  transition-timing-function: theme.transitionTiming;
  -webkit-transition-duration: theme.transitionDuration;
  transition-duration: theme.transitionDuration;
  -webkit-transition-property: color,border-color,background-color;
  transition-property: color,border-color,background-color;
  font-family: inherit;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 500;
  border-color: theme.mainRed;
  background-color: theme.mainRed;
  color: theme.greyWhite;
}

.c0:focus,
.c0:hover {
  border-color: theme.mainRed;
  color: theme.greyWhite;
}

<button
  className="c0"
/>
`;

exports[`Button Wrapper renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: theme.greyLight;
  border: 1px solid theme.greyMid;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition-timing-function: theme.transitionTiming;
  transition-timing-function: theme.transitionTiming;
  -webkit-transition-duration: theme.transitionDuration;
  transition-duration: theme.transitionDuration;
  -webkit-transition-property: color,border-color,background-color;
  transition-property: color,border-color,background-color;
  font-family: inherit;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 500;
}

.c0:focus,
.c0:hover {
  border-color: theme.mainRed;
  color: theme.greyWhite;
}

<button
  className="c0"
/>
`;

exports[`Button renders active correctly 1`] = `
<Wrapper
  active={true}
  onClick={[Function]}
>
  active button text
</Wrapper>
`;

exports[`Button renders correctly 1`] = `
<Wrapper
  active={false}
  onClick={[Function]}
>
  button label
</Wrapper>
`;

exports[`Button renders correctly with icon 1`] = `
<Wrapper
  active={false}
  onClick={[Function]}
>
  <IconWrapper>
    <MockIcon />
  </IconWrapper>
  icon button label
</Wrapper>
`;
